services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -157,11 +157,14 @@ public Response performLogin(AuthenticationRequest request) {
             Boolean allowCreate = null;
             if (getConfig().getConfig().get(SAMLIdentityProviderConfig.ALLOW_CREATE) == null || getConfig().isAllowCreate())
                 allowCreate = Boolean.TRUE;
+            Boolean forceAuthn = getConfig().isForceAuthn();
+            if ("login".equals(request.getAuthenticationSession().getClientNote(OIDCLoginProtocol.PROMPT_PARAM)))
```

Comment:
I would also check for the saml ForceAuthn note just in case keycloak is using SAML to login in the first place. It's the same but looking for [the note SamlProtocol.SAML_LOGIN_REQUEST_FORCEAUTHN](https://github.com/keycloak/keycloak/blob/23.0.4/services/src/main/java/org/keycloak/protocol/saml/SamlService.java#L494-L497). This way the identity provider forwards the re-login in both OIDC and SAML.